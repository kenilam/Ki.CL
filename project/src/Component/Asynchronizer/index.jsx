// @flow
import React from 'react';
import Async from 'react-async';

import { CSSTransition, Errors, Spinner } from 'Component';
import { debounce, isEmpty } from 'Helper';

import './style';

type Node = React.Node;

type AwaitForExpected = any;

type AwaitFor = Promise<AwaitForExpected>;

type Props = {
  children: Node,
  awaitFor: AwaitFor,
  awaitProps: {},
  awaitDelay?: Number,
  awaitMessage: String,
  iconOnly?: Boolean
};

const Component = ({ children, data }) =>
  typeof children === 'function'
    ? children({ data })
    : React.cloneElement(children, { data });

const Asynchronizer = ({
  children, // Accual children to be render
  awaitCache, // caches to await for
  awaitError,
  awaitFor, // function to await for
  awaitProps, // Props that pass to awaitFor
  awaitDelay,
  awaitMessage,
  iconOnly
}: Props) => {
  if (!isEmpty(awaitCache)) {
    return <Component data={awaitCache}>{children}</Component>;
  }

  return (
    <Async
      promiseFn={async () => {
        const data = await awaitFor(awaitProps);

        await debounce(awaitDelay);

        return data;
      }}
    >
      {({ data, error, isLoading }) => (
        <React.Fragment>
          <Spinner
            iconOnly={iconOnly}
            message={awaitMessage}
            show={Boolean(isLoading)}
          />
          <Errors errors={error} onError={awaitError} show={Boolean(error)} />
          <CSSTransition transitionIn={Boolean(data)}>
            <Component data={data}>{children}</Component>
          </CSSTransition>
        </React.Fragment>
      )}
    </Async>
  );
};

Asynchronizer.defaultProps = {
  awaitDelay: 1000,
  iconOnly: false
};

export default Asynchronizer;
