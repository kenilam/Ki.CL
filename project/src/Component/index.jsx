import GlobalHeader from './GlobalHeader';
import Link from './Link';
import Logo from './Logo';
import Nav from './Nav';
import Transition from './Transition';

export { GlobalHeader, Link, Logo, Nav, Transition };
