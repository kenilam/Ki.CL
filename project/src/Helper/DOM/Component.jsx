'use strict';

import AsyncEmitter from 'async-emitter';
import React from 'react/lib/ReactWithAddons';

import { ResizeSensor } from 'css-element-queries';
import { extend } from 'jquery';

import Unwrap from './Unwrap';

const Events = new AsyncEmitter();

class Component extends React.Component {
    static ResizeSensor = ResizeSensor;
    static Unwrap = Unwrap;
    static Events = Events;

    ResizeHandler = {
        add () {
            if (this.element && this.props.updateSizes) {
                this.updateSizes();

                new ResizeSensor(this.element, this.updateSizes);
            }
        },

        detect () {
            if (this.element && this.props.updateSizes) {
                ResizeSensor.detach(this.element, this.updateSizes);
            }
        }
    };

    ScrollHandler = {
        add () {
            if (this.props.scrollHandler || this.scrollHandler) {
                window.addEventListener('scroll', this.props.scrollHandler || this.scrollHandler);
            }
        },

        detect () {
            if (this.props.scrollHandler || this.scrollHandler) {
                window.removeEventListener('scroll', this.props.scrollHandler || this.scrollHandler);
            }
        }
    };

    constructor (props) {
        super(props);

        this.state = {};

        this.ScrollHandler.add = this.ScrollHandler.add.bind(this);
        this.ScrollHandler.detect = this.ScrollHandler.detect.bind(this);

        this.ResizeHandler.add = this.ResizeHandler.add.bind(this);
        this.ResizeHandler.detect = this.ResizeHandler.detect.bind(this);

        this.updateSizes = this.updateSizes.bind(this);
    }

    setState (currentState) {
        return new Promise(
            resolve => super.setState(
                previousState => extend(true, {}, previousState, currentState),
                () => resolve(currentState)
            )
        )
    }

    updateSizes () {
        cancelAnimationFrame(this.updateSizesFrame);

        this.updateSizesFrame = requestAnimationFrame(
            () => this.props.updateSizes({
                height: this.element.offsetHeight,
                width: this.element.offsetWidth
            })
        );
    }

    componentDidMount () {
        this.ScrollHandler.add();
        this.ResizeHandler.add();
    }

    componentWillUnmount () {
        this.ScrollHandler.detect();
        this.ResizeHandler.detect();
    }
}

export default Component;