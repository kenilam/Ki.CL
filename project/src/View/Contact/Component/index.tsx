import CTA from './CTA';
import Description from './Description';
import Title from './Title';

export { CTA, Description, Title };
