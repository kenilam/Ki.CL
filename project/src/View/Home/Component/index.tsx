import Phase from './Phase';
import Slogan from './Slogan';

export { Phase, Slogan };
