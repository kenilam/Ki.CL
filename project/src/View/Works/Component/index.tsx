import Background from './Background';
import Navigation from './Navigation';

export { Background, Navigation };
