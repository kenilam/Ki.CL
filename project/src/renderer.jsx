import React from 'react';
import ReactDOM from 'react-dom';

const appRoot = document.createDocumentFragment();

const renderer = App => {
    ReactDOM.render(<App />, appRoot);

    document.body.insertBefore(appRoot, document.body.childNodes[0]);
};

export default renderer;
