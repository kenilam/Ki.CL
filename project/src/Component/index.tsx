import Asynchronizer from './Asynchronizer';
import AudioPlayer from './AudioPlayer';
import CSSTransition from './CSSTransition';
import './Font';
import Link from './Link';
import Logo from './Logo';
import Navigation from './Navigation';
import Router from './Router';
import Transition from './Transition';
import WebGL from './WebGL';

export {
  Asynchronizer,
  AudioPlayer,
  CSSTransition,
  Link,
  Logo,
  Navigation,
  Router,
  Transition,
  WebGL
};
