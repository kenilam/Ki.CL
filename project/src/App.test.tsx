import * as React from 'react';
import renderer, {ReactTestRenderer, ReactTestRendererJSON} from 'react-test-renderer';
import Component from './App';

let component: ReactTestRenderer;
let instance: ReactTestRendererJSON;

describe('App', () => {
  beforeEach(() => {
    component = renderer.create(
      <Component />
    );
    
    instance = component.toJSON();
  });
  
  test('should match with snapshot', () => {
    expect(instance).toMatchSnapshot();
  });
});
