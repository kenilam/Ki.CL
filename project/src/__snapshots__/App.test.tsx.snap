// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match with snapshot 1`] = `undefined`;
